src/dune_rules/lib.ml

```diff
@@ -671,9 +671,11 @@ module Sub_system = struct
     | _ -> assert false
 
   let public_info lib =
-    Sub_system_name.Map.filter_map lib.sub_systems ~f:(fun (lazy inst) ->
-        let (Sub_system0.Instance.T ((module M), t)) = inst in
-        Option.map M.public_info ~f:(fun f -> M.Info.T (Result.ok_exn (f t))))
+    Result.try_with (fun () ->
```

Comment:
Is this meant to catch arbitrary exceptions, including things like `Code_error` raized by `f` or by forcing the lazy?
I wouldn't normally expect those to be caught, just as a general rule.

If the exceptions you're catching are just the ones raised by `Result.ok_exn` below, then we could use a locally-defined exception, or even no exception at all.

Reply:
Good call. I changed to only catch `User_error`. These are the only exceptions we are supposed to delay and is consistent with how we do this in `expander.ml`.

Reply:
Ok, seems fine.